src/lib/components/particles/waffle/index.jsx

```diff
@@ -0,0 +1,37 @@
+import defaultStyles from "./style.module.css"
+import { mergeStyles } from "$styles/helpers/mergeStyles"
+
+
+const renderUnit = (squares, attributes) => squares ? <rect {...attributes} /> : <circle {...attributes} />
+
+export const Waffle = ({ units, rows, total, squares = false, abbreviationAccessor = 'party', onMouseOver, paddingTop, showHalfLine, styles }) => {
+  //units is an array of objects with the following structure: { [abbreviationAccessor]: 'A' }.
+  const width = 1300
+  const columns = Math.ceil(total / rows)
+  const unitWidth = width / columns
+  const unitHeight = unitWidth
+  const height = rows * unitHeight + paddingTop
+  
+  styles = mergeStyles(defaultStyles, styles)
+
+  return (
+    <svg viewBox={`0 0 ${width} ${height}`} class={styles.svg}>
+      <g>
+        {
+          units.map((unit, j) => {
+            const attributes = squares ?
+            { onMouseOver: e => onMouseOver(unit, e), class: `${styles.unit} fill-color--${unit[abbreviationAccessor]}`, height: unitHeight, width: unitWidth, x: unitWidth * Math.floor(j / rows), y: unitHeight * (j % rows) + paddingTop} :
```

Comment:
See earlier comment about JSX readability. And to avoid election specific references, this could follow a similar pattern to how the map layers work, using `dynamicPropValue`, or each unit object could contain (an optional) CSS class.